core/rawdb/database.go

```diff
@@ -102,6 +102,79 @@ func NewDatabase(db ethdb.KeyValueStore) ethdb.Database {
 	}
 }
 
+// NewDatabaseWithFreezerRemote creates a high level database on top of a given key-
+// value data store with a freezer moving immutable chain segments into cold
+// storage.
+func NewDatabaseWithFreezerRemote(db ethdb.KeyValueStore, freezerURL string) (ethdb.Database, error) {
+	// Create the idle freezer instance
+	log.Info("New remote freezer", "freezer", freezerURL)
+
+	frdb, err := newFreezerRemoteClient(freezerURL)
+	if err != nil {
+		log.Error("NewDatabaseWithFreezerRemote error", "error", err)
+		return nil, err
+	}
+	// Since the freezer can be stored separately from the user's key-value database,
```

Comment:
We should update this comment to reflect what implementers should do or remove it altogether

Reply:
The commented code here differentiates core-geth from go-ethereum a little. Where go-ethereum won't attempt recovery in case of invalidation, core-geth will attempt to fix things. 

Although it would introduce a degree of variance, one idea here is to use the validate-only logic that go-ethereum uses. This will prevent mixing up freezers and KVs, while not implementing "fancy" recovery logic that may or may not be handled well by the remotes.

Reply:
See https://github.com/etclabscore/core-geth/pull/168/commits/6707b46d3ceb5da6bddb4472c11363519b745767